pages/settings.js

```diff
@@ -33,13 +58,19 @@ class SettingsForm extends React.Component {
     )
   }
 
-  submit(data) {
-    doApiRequest('/settings/?format=json', {
-      method: 'PATCH',
-      body: data,
-    }).then(resp => {
+  togglePublic(club) {
+    doApiRequest(
+      `/clubs/${club.code}/members/${this.props.userInfo.username}/?format=json`,
```

Comment:
can you put something like
```javascript
const { userInfo: { username }} = this.props
```
before this string interpolation to cut down on how long it is and make it a little easier to read?